botorch/acquisition/knowledge_gradient.py

```diff
@@ -428,16 +443,44 @@ def forward(self, X: Tensor) -> Tensor:
         return values.mean(dim=0)
 
 
+class ProjectedValueFunction(AcquisitionFunction):
+    r"""
+    Defines a wrapper around the value function that incorporates the project
+    operator.
+    """
+
+    def __init__(
+        self,
+        base_value_function: AcquisitionFunction,
+        project: Callable[[Tensor], Tensor],
+    ):
```

Comment:
Implied, but our convention is to add hints throughout.

```suggestion
    ) -> None:
```